### PR TITLE
V2 metrics

### PR DESCRIPTION
**Pull request by nagakingg**
### Description
This PR incorporates SimCurveCryptoPool into the metrics framework, including the StateLog and all pool-specific metrics. 

It addresses the item "metrics framework needs updating: getters, PoolBalance needs to incorporate price, PoolValue needs to use xcp(D)" in #96 

Specific changes/additions:
- incorporate SimCurveCryptoPool into required metrics
- add tests for get_pool_parameters, get_pool_state, each metric class
- metrics.state_log: Add get_pool_parameters and get_pool_state functions for SimCurveCryptoPool

### Note
I added basic tests for each metric to ensure that all the base class properties/attributes/methods work correctly. I did not add unit tests for any of the specific metrics in metrics.metrics, although these are exercised in the end-to-end tests. Those will take some further thought/discussion on how best to test without simply duplicating the code. For the pool-specific CryptoPool metrics, running some simulations may help to reveal any issues.

### Hygiene checklist

- [x] Changelog entry
- [x] Everything public has a Numpy-style docstring
      (modules, public functions, classes, and public methods)
- [x] Pylint score monotonically increased
- [x] Commit history is cleaned-up with minor changes squashed together
      and descriptive commit messages following [Tim Pope's style](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: curvesim/metrics/base.py [hunk 1]
```diff
@@ -356,6 +351,22 @@ def get_market_price(self, base, quote, prices):
...
 
 
+def get_coin_pairs(prices):
+    """
+    Returns the coin pairs available in the price data.
+    """
+    if isinstance(prices, DataFrame):
+        return prices.columns
+    if isinstance(prices, Series):
+        return prices.index
```
Comment (chanhosuh) [inline curvesim/metrics/base.py:361]: Should make sure we return the same type for each option

File: test/unit/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+def test_get_pool_parameters_unmapped_exception():
+    """Test that unmapped pool throws StateLogError."""
+
+    class DummyPool:
+        pass
+
+    pool = DummyPool()
+
+    with pytest.raises(StateLogError):
```
Comment (chanhosuh) [inline test/unit/test_metrics.py:57]: would probably be better to raise a more specific error...

class UnregisteredPoolError(StateLogError):
    pass

**Review by chanhosuh — APPROVED** [review]

File: curvesim/metrics/base.py (hunk 1 above)
Comment (nagakingg) [reply curvesim/metrics/base.py]: All return tuples now. Added an explicit test that the same data and type are returned regardless of input format in test_metrics.test_get_poin_pairs

File: test/unit/test_metrics.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    pools = [
+        sim_curve_pool,
+        sim_curve_crypto_pool,
+        sim_curve_meta_pool,
+        sim_curve_rai_pool,
+    ]
+
+    for pool in pools:
+        params = get_pool_parameters(pool)
+        assert params == pool_parameter_functions[type(pool)](pool)
```
[the remark this answers is not in the document]
Comment (nagakingg) [reply test/unit/test_metrics.py]: Changed these to explicitly test the output. There is actually just enough logic in these functions so the tests don't simply duplicate the code. I moved these to test_state_log

File: test/unit/test_metrics.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    pools = [
+        sim_curve_pool,
+        sim_curve_crypto_pool,
+        sim_curve_meta_pool,
+        sim_curve_rai_pool,
+    ]
+
+    for pool in pools:
+        state = get_pool_state(pool)
+        assert state == pool_state_functions[type(pool)](pool)
```
[the remark this answers is not in the document]
Comment (nagakingg) [reply test/unit/test_metrics.py]: Also moved to test_state_log

File: test/unit/test_metrics.py (hunk 2 above)
Comment (nagakingg) [reply test/unit/test_metrics.py]: Good call. Changed

**Pull request merged**